clients/drcachesim/scheduler/scheduler_impl.cpp

```diff
@@ -2673,6 +2762,26 @@ scheduler_impl_tmpl_t<RecordType, ReaderType>::next_record(output_ordinal_t outp
     outputs_[output].last_record = record;
     record_type_has_tid(record, input->last_record_tid);
     record_type_has_pid(record, input->pid);
+
+    trace_marker_type_t marker_type;
+    uintptr_t marker_value;
+    // Good to queue the injected records at this point, because we now surely will
+    // be done with TRACE_MARKER_TYPE_SYSCALL.
```

Comment:
Can this be moved inside process_marker()?  It feels out of place here.  If for some reason it has to be here: please make a new helper function as this containing next_record() function is already long.

Reply:
> make a new helper function as this containing next_record() function is already long.

Added a finalize_next_record which can hold these "final" actions.